### PR TITLE
HMAN-45 Map delete request to HMI

### PR DESCRIPTION
**Pull request by markdathornehmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/HMAN-45

### Change description ###

Map delete request to DELETE request to HMI payload for HMI

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java [hunk 1]
```diff
@@ -121,4 +126,13 @@ public void updateHearing(@RequestBody @Valid UpdateHearingRequest hearingReques
...
     }
+
+
+    @PutMapping(path = "/test/{id}", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
+    public HmiSubmitDeleteHearingRequest test(@RequestBody @Valid DeleteHearingRequest hearingRequest,
+                                              @PathVariable("id") Long hearingId) {
+        hearingManagementService.deleteHearingRequest(hearingId, hearingRequest);
+        return hearingManagementService.test(hearingRequest);
+    }
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java:137]: Just a note to remember to delete before merging (no action required right now)

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java [hunk 2]
```diff
@@ -36,4 +46,20 @@ public HmiSubmitHearingRequest mapRequest(Long hearingId, HearingRequest hearing
             .build();
     }
 
+    /**
+     * map Delete hearing Request to HMI Submit Delete Hearing Request.
+     * @param hearingRequest delete hearing request
+     * @return hmiSubmitDeleteHearingRequest HMI Submit Delete Hearing request object
+     */
+    public HmiSubmitDeleteHearingRequest mapRequest(DeleteHearingRequest hearingRequest) {
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java:54]:
```suggestion
    public HmiDeleteHearingRequest mapRequest(DeleteHearingRequest hearingRequest) {
```
Can we also move this to a separate class please as this class is for mapping Submit Hearing requests. `HmiDeleteHearingRequestMapper` would be more appropriate for this method to sit in

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java [hunk 3]
```diff
@@ -21,6 +25,12 @@ public HmiSubmitHearingRequestMapper(HmiCaseDetailsMapper hmiCaseDetailsMapper,
         this.listingMapper = listingMapper;
     }
 
+    /**
+     * map hearing Request to HMI Submit Hearing Request.
+     * @param hearingId haring id
+     * @param hearingRequest hearing request
+     * @return hmiSubmitHearingRequest HMI Submit Hearing request object
+     */
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java:33]: This (and the other JavaDoc) doesn't really add any benefit; I would just delete it as the method signature is already clear enough.

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java [hunk 4]
```diff
@@ -36,4 +46,20 @@ public HmiSubmitHearingRequest mapRequest(Long hearingId, HearingRequest hearing
...
+     * @return hmiSubmitDeleteHearingRequest HMI Submit Delete Hearing request object
+     */
+    public HmiSubmitDeleteHearingRequest mapRequest(DeleteHearingRequest hearingRequest) {
+
+        HmiDeleteHearingRequest hmiDeleteHearingRequest = HmiDeleteHearingRequest.builder()
+                .cancellationReason(new CancellationReason(hearingRequest.getCancellationReasonCode()))
+                .build();
+
+        return HmiSubmitDeleteHearingRequest.builder()
+                .hearingRequest(hmiDeleteHearingRequest)
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java:61]: `hearingRequest` is not part of the HMI spec for Delete, only Submit. The example request is
```
{
    "_cancel": {
        "cancellationReasonCode": "CNCL"
    }
}  
```

whereas this generates
```
{
    "hearingRequest": {
        "_cancel": {
            "cancellationReasonCode": "CNCL"
        }
    } 
}  
```

It looks like we can probably just return the previously built `HmiDeleteHearingRequest` and delete this extra `HmiSubmitDeleteHearingRequest` object completely and it'll give the desired result.

**Review by danlysiak — COMMENTED** [review]
Just a couple of comments please

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java (hunk 3 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java:29]: Removing now

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java (hunk 2 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiSubmitHearingRequestMapper.java]: Done

**Review by danlysiak — APPROVED** [review]

**Pull request merged**